express-notes-api/server.js

```diff
@@ -0,0 +1,145 @@
+import express from 'express';
+import * as fs from 'node:fs/promises';
+
+const app = express();
+const jsonParser = express.json();
+
+app.use(jsonParser);
+
+app.use('/api/notes/:id', (req, res, next) => {
+  if (!Number(req.params.id)) {
+    res.status(400).send({ error: `id ${req.params.id} must be a positive integer.` });
+  } else {
+    next();
+  }
+});
+
+app.get('/api/notes', async (req, res) => {
+  const theObject = await read();
+  const notesList = [];
+  for (const [, value] of Object.entries(theObject.notes)) {
+    notesList.push(value);
+  }
+  res.json(notesList);
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const theObject = await read();
+  for (const [key, value] of Object.entries(theObject.notes)) {
+    if (req.params.id === key) {
+      res.json(value);
+      return;
+    }
+  }
+  res.status(400).send(`Id ${req.params.id} not found.`);
```

Comment:
One rule of error handling in express is that if an exception bubbles outside of an async route handler, express will crash. The best way to handle this is to put a try/catch around every async route handler. If you do this, you can eliminate the try/catch in most of your other code, knowing that it will be taken care of in the route-handler.